frontend/src/metabase/visualizations/echarts/cartesian/model/index.ts

```diff
@@ -97,8 +97,13 @@ export const getCartesianChartModel = (
   }
   dataset = sortDataset(dataset, settings["graph.x_axis.scale"]);
 
+  const xAxisSeriesModel = seriesModels.find(
+    seriesModel => seriesModel.column.name === dimensionModel.column.name,
+  );
+
   const xAxisModel = getXAxisModel(
     dimensionModel,
+    xAxisSeriesModel,
```

Comment:
@EmmadUsmani @alxnddr does it look alright to pass the series model here? I have a strange gut feeling, haha

Reply:
Also, I made the fix on the "chart model" level if that makes sense, but we also seem to have some label logic when building the echarts option object (e.g. [here](https://github.com/metabase/metabase/blob/e7f80a90ab05cca699e49acfeb48733e17a9f43d/frontend/src/metabase/visualizations/echarts/cartesian/option/axis.ts#L153)). Not sure if we need to update that as well? Do we need to specify labels in both chart model and option object at all?